crates/sui-framework/packages/deepbook/sources/clob_v2.move

```diff
@@ -1257,6 +1422,69 @@ module deepbook::clob_v2 {
         account_cap: &AccountCap,
         ctx: &mut TxContext
     ): (u64, u64, bool, u64) {
+        let (base_quantity_filled, quote_quantity_filled, is_success, order_id, _meta_data) = place_limit_order_int(
+            pool,
+            client_order_id,
+            price,
+            quantity,
+            self_matching_prevention,
+            is_bid,
+            expire_timestamp, // Expiration timestamp in ms in absolute value inclusive.
+            restriction,
+            clock,
+            account_cap,
+            ctx
+        );
+        (base_quantity_filled, quote_quantity_filled, is_success, order_id)
+    }
+
+    public fun place_limit_order_with_metadata<BaseAsset, QuoteAsset>(
+        pool: &mut Pool<BaseAsset, QuoteAsset>,
+        client_order_id: u64,
+        price: u64,
+        quantity: u64,
+        self_matching_prevention: u8,
+        is_bid: bool,
+        expire_timestamp: u64, // Expiration timestamp in ms in absolute value inclusive.
+        restriction: u8,
+        clock: &Clock,
+        account_cap: &AccountCap,
+        ctx: &mut TxContext
+    ): (u64, u64, bool, u64, vector<MatchedOrderMetadata<BaseAsset, QuoteAsset>>) {
+        let (base_quantity_filled, quote_quantity_filled, is_success, order_id, meta_data) = place_limit_order_int(
+            pool,
+            client_order_id,
+            price,
+            quantity,
+            self_matching_prevention,
+            is_bid,
+            expire_timestamp, // Expiration timestamp in ms in absolute value inclusive.
+            restriction,
+            clock,
+            account_cap,
+            ctx
+        );
+        (base_quantity_filled, quote_quantity_filled, is_success, order_id, meta_data)
+    }
+
+    /// Place a limit order to the order book.
+    /// Returns (base quantity filled, quote quantity filled, whether a maker order is being placed, order id of the maker order).
+    /// When the limit order is not successfully placed, we return false to indicate that and also returns a meaningless order_id 0.
+    /// When the limit order is successfully placed, we return true to indicate that and also the corresponding order_id.
+    /// So please check that boolean value first before using the order id.
```

Comment:
let's remove the comment here and add one to the public function with_metadata

Reply:
removed function desc from internal func and keeping it for public fun.